src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -132,17 +131,120 @@ def __init__(self, *args, **kwargs):
+        try:
+            with open(os.path.join(os.sep, "scratch",
+                                   "CI-skip-list-master")) as skip_handle:
+                skip_list = skip_handle.readlines()
+        except Exception as excpt: # pylint: disable=broad-except
+            skip_process_error("Unable to read skip list: {}".format(excpt))
+            skip_list = []
+
+        for item in skip_list:
+            vals = item.split('|')
+            skip_it, ticket = self._check_variant_skip(literal_eval(vals[0]))
+            if skip_it:
+                # test is on the skiplist
+                if len(vals) > 1:
+                    # but there is a commit that fixes it
+                    try:
+                        with open(os.path.join(os.sep, 'tmp',
+                                               'commit_list')) as commit_handle:
+                            commits = commit_handle.readlines()
+                    except Exception as excpt: # pylint: disable=broad-except
+                        skip_process_error("Unable to read commit list: "
+                                           "{}".format(excpt))
+                        commits = None
+                    if commits and vals[1] in commits:
+                        # fix is in this code base
+                        self.log.info("This test variant is included in the "
+                                     "skip list for ticket {}, but is fixed "
+                                     "in {}.  Test will not be "
+                                     "skipped".format(ticket, vals[1]))
+                    else:
+                        # fix is not in this code base
+                        self.log.info("Skipping due to being on the "
+                                      "skip list for ticket {}, and "
+                                      "the fix in {} is not in the "
+                                      "current code".format(ticket,
+                                                            vals[1]))
+                        cancelForTicket(ticket, skip_list)
+                else:
+                    try:
+                        with open(os.path.join(os.sep, 'tmp',
+                                  'commit_title')) as commit_handle:
+                            if commit_handle.read().strip().startswith(
+                                    ticket + " "):
+                                # fix is in this PR
+                               self.log.info("This test variant is included in "
+                                             "the skip list for ticket {}, but "
+                                             "it is being fixed in this PR.  "
+                                             "Test will not be "
+                                             "skipped".format(ticket))
+                            else:
+                                # there is no commit that fixes it
+                                self.log.info("This test variant is included "
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions